pkg/kubectl/cmd/create.go

```diff
@@ -106,14 +109,59 @@ func NewCmdCreate(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
 	return cmd
 }
 
-func ValidateArgs(cmd *cobra.Command, args []string) error {
+func (o *CreateOptions) ValidateArgs(cmd *cobra.Command, args []string) error {
 	if len(args) != 0 {
 		return cmdutil.UsageErrorf(cmd, "Unexpected args: %v", args)
 	}
+	if len(o.Raw) > 0 {
+		if o.EditBeforeCreate {
+			return cmdutil.UsageErrorf(cmd, "--raw and --edit are mutually exclusive")
+		}
+		if len(o.FilenameOptions.Filenames) != 1 {
+			return cmdutil.UsageErrorf(cmd, "--raw can only use a single local file or stdin")
+		}
+		if strings.HasPrefix(o.FilenameOptions.Filenames[0], "http") {
+			return cmdutil.UsageErrorf(cmd, "--raw can only use a single local file or stdin")
+		}
+		if o.FilenameOptions.Recursive {
+			return cmdutil.UsageErrorf(cmd, "--raw and --recursive are mutually exclusive")
+		}
+		if len(o.Selector) > 0 {
+			return cmdutil.UsageErrorf(cmd, "--raw and --recursive are mutually exclusive")
+		}
+	}
+
 	return nil
 }
 
 func RunCreate(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, options *CreateOptions) error {
+	// raw only makes sense for a single file resource multiple objects aren't likely to do what you want.
+	// the validator enforces this, so
+	if len(options.Raw) > 0 {
+		restClient, err := f.RESTClient()
+		if err != nil {
+			return err
+		}
+
+		var data io.ReadCloser
+		if options.FilenameOptions.Filenames[0] == "-" {
+			data = os.Stdin
+		} else {
+			data, err = os.Open(options.FilenameOptions.Filenames[0])
+			if err != nil {
+				return err
+			}
+		}
+		// TODO post content with stream.  Right now it ignores body content
+		bytes, err := restClient.Post().RequestURI(options.Raw).Body(data).DoRaw()
```

Comment:
How does the error look like if we have an invalid URI here (e.g. a full URL, non-existent URI)? User-friendly enough expose it directly?

Reply:
Agree here.

If we have an invalid URI here, panic will occur instead of error.

Reply:
Do we need to make it clear somewhere (e.g. the flag description) that the output when using `--raw` is always JSON? Does it make it mutually exclusive with `--output` and other printer flags?